File: ProjetoJava/src/classeAbstrata.java
abstract class Funcionario {
    abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private double horasTrabalhadas;
    private double valorHora;

    public FuncionarioHorista(double horasTrabalhadas, double valorHora) {
        this.horasTrabalhadas = horasTrabalhadas;
        this.valorHora = valorHora;
    }

    @Override
    public double calcularSalario() {
        return horasTrabalhadas * valorHora;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salarioMensal;

    public FuncionarioAssalariado(double salarioMensal) {
        this.salarioMensal = salarioMensal;
    }

    @Override
    public double calcularSalario() {
        return salarioMensal;
    }

    public static void main(String[] args) {
        FuncionarioHorista horista = new FuncionarioHorista(160, 25);
        FuncionarioAssalariado assalariado = new FuncionarioAssalariado(5000);

        System.out.println("Salário horista: R$" + horista.calcularSalario());
    }
}

File: GoLangProject/classeAbstrata.go
package main

import "fmt"

type Funcionario interface {
	CalcularSalario() float64
}

type FuncionarioHorista struct {
	HorasTrabalhadas int
	ValorHora        float64
}

func (f FuncionarioHorista) CalcularSalario() float64 {
	return float64(f.HorasTrabalhadas) * f.ValorHora
}

type FuncionarioAssalariado struct {
	SalarioMensal float64
}

func (f FuncionarioAssalariado) CalcularSalario() float64 {
	return f.SalarioMensal
}

func main() {
	horista := FuncionarioHorista{160, 25}
	assalariado := FuncionarioAssalariado{5000}

	fmt.Printf("Salário horista: R$%.2f\n", horista.CalcularSalario())
	fmt.Printf("Salário assalariado: R$%.2f\n", assalariado.CalcularSalario())
}
